clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -534,30 +524,29 @@ private void handleCompletedReceives(List<ClientResponse> responses, long now) {
             if (req.isInternalRequest && body instanceof MetadataResponse)
                 metadataUpdater.handleCompletedMetadataResponse(req.header, now, (MetadataResponse) body);
             else if (req.isInternalRequest && body instanceof ApiVersionsResponse)
-                handleApiVersionsResponse(req, (ApiVersionsResponse) body);
+                handleApiVersionsResponse(responses, req, now, (ApiVersionsResponse) body);
             else
                 responses.add(req.completed(body, now));
         }
     }
 
-    private void handleApiVersionsResponse(InFlightRequest req, ApiVersionsResponse apiVersionsResponse) {
+    private void handleApiVersionsResponse(List<ClientResponse> responses,
+            InFlightRequest req, long now, ApiVersionsResponse apiVersionsResponse) {
         final String node = req.destination;
-        for (ApiVersionsResponse.ApiVersion requiredApiVersion : requiredApiVersions) {
-            final ApiVersionsResponse.ApiVersion supportedApiVersion = apiVersionsResponse.apiVersion(requiredApiVersion.apiKey);
-            if (supportedApiVersion == null) {
-                close(node);
-                throw new KafkaException("Node " + req.destination + " does not support Api " + requiredApiVersion.apiKey + ".");
-            }
-            if (supportedApiVersion.maxVersion < requiredApiVersion.minVersion || supportedApiVersion.minVersion > requiredApiVersion.maxVersion) {
-                close(node);
-                throw new KafkaException("Node " + req.destination + " does not support required versions for Api " + requiredApiVersion.apiKey + "." +
-                        " Supported versions: " + "[" + supportedApiVersion.minVersion + ", " + supportedApiVersion.maxVersion + "]" + "," +
-                        " required versions: " + "[" + requiredApiVersion.minVersion + ", " + requiredApiVersion.maxVersion + "]" + ".");
-            }
+        if (apiVersionsResponse.errorCode() != Errors.NONE.code()) {
+            log.warn("Node " + node + " got error " + apiVersionsResponse.errorCode() +
```

Comment:
We have the following code for failed metadata requests:

```java
          if (cluster.isBootstrapConfigured()) {
                int nodeId = Integer.parseInt(destination);
                Node node = cluster.nodeById(nodeId);
                if (node != null)
                    log.warn("Bootstrap broker {}:{} disconnected", node.host(), node.port());
            }
```

Should we be doing similarly for this case (since we now do `ApiVersionsRequest` before `MetadataRequest`)?